### ISSUE TITLE
how packets go back from outside of cluster when using ipvlan technology

### ISSUE THREAD

**Issue by bretagne-peiqi** [cilium/cilium#8885]
## Support

I am experiencing with cilium 1.4.6 using ipvlan technology with direct-route-mode. 

We are not supposed to config ToR or Leaf physical Swithes.  when packets go out of cluster, they can't find way back.  Is there a way for cilium like calico using snat on node host. but more performanent ? 

or we have to config ToR if using this tech ?

**Comment by joestringer** [comment]
Have you tried enabling the `masquerade` option in Cilium?

**Comment by bretagne-peiqi** [comment]
no,  it works as what calico does ?

**Comment by bretagne-peiqi** [comment]
I enabled masquerade, it works partially.
 I think cilium should have a more intelligent method to select network
outside cluster from those inside, typically if can't find a direct route
and  go out through default route, we should masquerade it,  whatever they
are all in 10.x.x.x pattern.

**Comment by bretagne-peiqi** [comment]
<img width="1433" alt="Screen Shot 2019-08-14 at 1 53 57 PM" src="https://user-images.githubusercontent.com/5621794/62998230-7430aa00-be9d-11e9-97f3-98669708e5fe.png">

I changed cluster CIDR from automatic generated cider to specifying in kube-controller-manager. specifically, 10.x.x.x to 192.169.0.0/16 to avoid redundancy.

however, It happens the ip range  full error ..

**Comment by tgraf** [comment]
@bretagne-peiqi If you run with `--tunnel=disabled` Cilium will always defer to the Linux routing layer and use whatever route is configured. Is this what you want?

The SNAT rule currently only applies to traffic that leaves the cluster prefix as this is what the majority of users wants, it's trivial to add a flag to enable SNAT for all traffic if that is what you want.

**Comment by bretagne-peiqi** [comment]
yes, I disabled tunnel and enable masquerade.  we have two k8s cluster, 10.130.0.0/16 and 10.128.0.0/16.  In 10.128 cluster, I deployed cilium and cidr is derived from node ip. 
after enable masquerade, I can ping 114.114.114.114 reachable. but if I ping 10.130.x.x, it is not reachable, and from the trace, I saw src ip is still the pod ipv4. I am wondering cilium can't figure out 10.130.x.x as external cidr.
